packages/main/src/Dialog.js

```diff
@@ -286,6 +301,21 @@ class Dialog extends Popup {
 		return minHeight;
 	}
 
+	get hasValueState() {
+		return this.state !== ValueState.None;
+	}
+
+	get _dialogStateIcon() {
+		const iconPerValueState = {
```

Comment:
Extract this object to an outer scope to avoid reinitializing it every time function is called.